tests/helper_functions.py

```diff
@@ -817,3 +818,24 @@ def get_logged_model_by_name(name: str) -> Optional[LoggedModel]:
         filter_string=f"name='{name}'", output_format="list", max_results=1
     )
     return logged_models[0] if len(logged_models) >= 1 else None
+
+
+def wait_for_telemetry_threads(client: TelemetryClient = None, terminate: bool = False):
+    """Wait for telemetry threads to finish to avoid race conditions in tests.
+
+    Args:
+        client: The telemetry client to wait for. If None, uses the global client.
+        terminate: If True, terminates the threads after flushing.
+    """
+    telemetry_client = client or get_telemetry_client()
+    if telemetry_client is None:
+        return
+
+    # Flush the telemetry client to ensure all pending records are processed
+    telemetry_client.flush(terminate=terminate)
+
+    if terminate:
+        # Wait for threads to finish -- consumer threads will be terminated
+        for thread in telemetry_client._consumer_threads:
+            if thread.is_alive():
+                thread.join(timeout=1)
```

Comment:
can we add this as an instance method of `wait_for_telemetry_threads`? Having a separate function is inconvenient because you need to import it.